spectrum/src/main/java/io/github/giulong/spectrum/interfaces/WebElementFinder.java

```diff
@@ -0,0 +1,13 @@
+package io.github.giulong.spectrum.interfaces;
+
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+public interface WebElementFinder {
```

Comment:
- [ ] As it is now, this interface is a bit useless, because you're always referencing the concrete implementation (`LocatorType`) when you need this, as you can see in every `Js` method, for example:
  ```java
  public List<WebElement> findElements(final LocatorType locatorType // <---
  ```
  This violates the [Dependency Inversion Principle](https://en.wikipedia.org/wiki/Dependency_inversion_principle) (the 'D' in 'SOLID'). You should replace all those with the interface:
  ```java
  public List<WebElement> findElements(final WebElementFinder webElementFinder
  ```

Reply:
Updated on commit: [d05f3b3](https://github.com/giulong/spectrum/pull/159/commits/d05f3b3bcef59a2c78aa06520a9b13a44645b590) as suggested.